### PR TITLE
HITL - Configure camera height based on the agent height.

### PR DESCRIPTION
**Pull request by 0mdc**
## Motivation and Context

This changeset sets the HITL first person camera height from the agent type they are controlling.

This is hacky - This will be moved into a purposeful API in the HITL controllers.

## How Has This Been Tested

Tested locally.

## Types of changes

- **\[Development\]**

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: examples/hitl/rearrange_v2/rearrange_v2.py [hunk 1]
```diff
@@ -365,11 +367,32 @@ def pop_ui_events(self) -> List[Dict[str, Any]]:
         return events
 
     def _get_camera_lookat_pos(self) -> mn.Vector3:
+        # HACK: Estimate camera height.
+        # TODO: GuiController should have knowledge on the agent type it controls, and should provide an API to get the camera height.
+        gui_agent_controller = self.gui_agent_controller
+        sim = self.app_service.sim
+        agent_data = self.agent_data
+        camera_height_offset = 1.0
+        if isinstance(gui_agent_controller, GuiRobotController):
```
Comment (0mdc) [inline examples/hitl/rearrange_v2/rearrange_v2.py:376]: @zephirefaith : Beware that until this code is handled at the right place, you'll have to check for `LLMController`.

File: examples/hitl/rearrange_v2/rearrange_v2.py [hunk 2]
```diff
@@ -365,11 +367,32 @@ def pop_ui_events(self) -> List[Dict[str, Any]]:
...
+            # The robot camera height is defined in `spot_robot.py`.
+            # It's directly attached to the 6th articulated object link, without a height offset.
+            agent = sim.agents_mgr[agent_data.agent_index].articulated_agent
+            root = agent.sim_obj
+            head_link = root.get_link_scene_node(6)
+            head_link_height = head_link.absolute_translation.y
+            base_link = root.get_link_scene_node(-1)
+            base_link_height = base_link.absolute_translation.y
+            camera_height_offset = head_link_height - base_link_height
+        elif isinstance(gui_agent_controller, GuiHumanoidController):
```
Comment (zephirefaith) [inline examples/hitl/rearrange_v2/rearrange_v2.py:386]: I think I can just expand this condition to trigger when `gui_agent_controller` is `LLMController`

**Review by zephirefaith — APPROVED** [review]
Makes sense. Thanks for figuring this out!

File: examples/hitl/rearrange_v2/rearrange_v2.py (hunk 2 above)
Comment (0mdc) [reply examples/hitl/rearrange_v2/rearrange_v2.py:386]: Correct!
Eventually, the controller will know about the agent it's controlling and will be able to get the correct camera height.
Better have this hack in and expose the complexity for now.

**Pull request merged**